modules/db/src/main/Handlers.scala

```diff
@@ -17,22 +17,27 @@ trait Handlers:
 
   // free handlers for all types with TotalWrapper
   // unless they are given an instance of lila.db.NoDbHandler[T]
-  given opaqueHandler[T, A](using
-      sr: SameRuntime[A, T],
+  given opaqueWriter[T, A](using
       rs: SameRuntime[T, A],
-      handler: BSONHandler[A]
-  )(using NotGiven[NoDbHandler[T]]): BSONHandler[T] =
-    handler.as(sr.apply, rs.apply)
+      writer: BSONWriter[A]
+  )(using NotGiven[NoDbHandler[T]]): BSONWriter[T] with
+    def writeTry(t: T) = writer.writeTry(rs(t))
 
-  given listHandler[T: BSONHandler]: BSONHandler[List[T]] with
```

Comment:
From https://github.com/scala/scala3/pull/19300, compiler will pick the most general given instance: https://github.com/scala/scala3/pull/19300/files#diff-b961213fc5ea62f83ece999f7275ee195fed5a57e4f1c4e6b39dc9d7232a7a90R169. So listHandler will never be picked over reactivemongo collectionWriter. Similar for `vectorHandler`.